### PR TITLE
Soft assert cleanup

### PR DESCRIPTION
**Pull request by czue**
@benrudolph @esoergel

the first one never fires and the second one is not providing any value anymore.

### CODE REVIEW

**Comment by czue** [comment]
(good to wait for tests just in case)

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 1]
```diff
@@ -300,10 +300,11 @@ class RestoreState(object):
     restore_class = FileRestoreResponse
 
     def __init__(self, project, user, params):
+        if not project or not project.name:
+            raise Exception('you are not allowed to make a RestoreState without a domain!')
```
Comment (esoergel) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: :+1:

**Comment by esoergel** [comment]
looks g2m after build

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**